haskell/private/haskell_impl.bzl

```diff
@@ -233,10 +233,17 @@ def _haskell_binary_common_impl(ctx, is_test):
         version = ctx.attr.version,
     )
 
+    boot_files = depset([
+        s
+        for s in c.extra_source_files.to_list()
+        if s.extension in ["hs-boot", "lhs-boot"]
+    ])
```

Comment:
Can we push the separation all the way through? I.e. already have a separate `boot_files` returned by `compile_binary` and not include the boot files in `c.extra_source_files` in the first place? That would seem like a more consistent implementation.

Reply:
Please, take a look at my attempt. I tried not to forget every `extra_source_files` usage and provide the boot files.